src/js_printer.zig

```diff
@@ -2045,23 +2282,65 @@ pub fn NewPrinter(
                         }
                     }
 
-                    p.print("`");
-                    if (e.head.isPresent()) {
-                        e.head.resovleRopeIfNeeded(p.options.allocator);
+                    var is_ascii = e.head.isUTF8();
 
-                        p.printStringContent(&e.head, '`');
+                    for (e.parts) |*part| {
+                        is_ascii = is_ascii and part.tail.isUTF8();
                     }
 
-                    for (e.parts) |*part| {
-                        p.print("${");
-                        p.printExpr(part.value, .lowest, ExprFlag.None());
-                        p.print("}");
-                        if (part.tail.isPresent()) {
+                    if (e.is_raw_template_call and !is_ascii) {
+                        p.print("(Object.freeze(Object.defineProperty([");
```

Comment:
This logic should probably be moved to somewhere in js_parser, not just for formatting but so can place this in a hoisted variable.

E.g. 
```js
function f() {
    return foo`💩`
}
```
should transpile to:
```js
const regexp_1 = Object.freeze(Object.defineProperty(["\u{1F4A9}"], "raw", { value: Object.freeze(["\u{1F4A9}"]) }));

function f() {
    return foo(regexp_1);
}
```